src/ethereum_optimized/state_db.py

```diff
@@ -250,6 +301,30 @@ def rollback_transaction(state: State) -> None:
         if not state.tx_restore_points:
             state.created_accounts.clear()
 
+    if has_transient_storage:
+
+        @add_item(patches)
+        def rollback_transaction(
+            state: State,
+            transient_storage: Optional[Any] = None,
+        ) -> None:
```

Comment:
The original function has a signature of:

https://github.com/ethereum/execution-specs/blob/11bb742f063335d1c94878356d02bf49cafb08d4/src/ethereum/cancun/state.py#L121-L123

Do we still need the `Optional` in the patch?

Reply:
No, but it's harmless.